modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-test/src/testIntegration/java/com/liferay/headless/commerce/admin/catalog/resource/v1_0/test/ProductOptionResourceTest.java

```diff
@@ -6,15 +6,354 @@
 package com.liferay.headless.commerce.admin.catalog.resource.v1_0.test;
 
 import com.liferay.arquillian.extension.junit.bridge.junit.Arquillian;
+import com.liferay.commerce.product.model.CPDefinition;
+import com.liferay.commerce.product.model.CPInstance;
+import com.liferay.commerce.product.model.CPOption;
+import com.liferay.commerce.product.model.CProduct;
+import com.liferay.commerce.product.test.util.CPTestUtil;
+import com.liferay.headless.commerce.admin.catalog.client.dto.v1_0.ProductOption;
+import com.liferay.headless.commerce.admin.catalog.client.pagination.Page;
+import com.liferay.headless.commerce.core.util.LanguageUtils;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.util.GetterUtil;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.odata.entity.EntityField;
 
+import java.math.BigDecimal;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.junit.Assert;
+import org.junit.Before;
 import org.junit.Ignore;
+import org.junit.Test;
 import org.junit.runner.RunWith;
 
 /**
- * @author Zoltán Takács
+ * @author Stefano Motta
  */
-@Ignore
 @RunWith(Arquillian.class)
 public class ProductOptionResourceTest
 	extends BaseProductOptionResourceTestCase {
+
+	@Before
+	@Override
+	public void setUp() throws Exception {
+		super.setUp();
+
+		_cpInstance = CPTestUtil.addCPInstanceWithRandomSku(
+			testGroup.getGroupId(), BigDecimal.TEN);
+
+		_cpDefinition = _cpInstance.getCPDefinition();
+
+		_cProduct = _cpDefinition.getCProduct();
+	}
+
+	@Ignore
+	@Override
+	@Test
+	public void testDeleteProductOption() throws Exception {
+		super.testDeleteProductOption();
+	}
+
+	@Ignore
+	@Override
+	@Test
+	public void testGraphQLDeleteProductOption() throws Exception {
+		super.testGraphQLDeleteProductOption();
+	}
+
+	@Override
+	@Test
+	public void testPatchProductOption() throws Exception {
+		ProductOption postProductOption = _postProductIdProductOptionsPage();
```

Comment:
not sure about the method name. at line 99 there's a method that actually return a Page obj while here it is not, can you please check?